### ISSUE TITLE
LG ThinQ internal server error, try again later.

### ISSUE THREAD

**Issue by xcsdm** [nVuln/homebridge-lg-thinq#118]
**Describe The Bug:**
Cannot connect to LG servers.  Error in HomeBridge log:

[1/10/2022, 8:23:31 AM] [LGThinQ] LG ThinQ internal server error, try again later.
[1/10/2022, 8:23:31 AM] [LGThinQ]

**To Reproduce:**
I installed the plugin fresh.  I was able to authenticate, and receive a refresh token.

**Expected behavior:**
Unknown.  I have never seen it work.

**Logs:**

```
[1/10/2022, 8:23:31 AM] [LGThinQ] LG ThinQ internal server error, try again later.
[1/10/2022, 8:23:31 AM] [LGThinQ]
```

**Plugin Config:**

```
 {
            "country": "US",
            "language": "en-US",
            "auth_mode": "token",
            "refresh_token": "<REMOVED>",
            "username": "",
            "password": "",
            "platform": "LGThinQ",
            "devices": [],
            "thinq1": false,
            "refresh_interval": 60
        }
```

**Screenshots:**

**Environment:**

* **Plugin Version**:
* **Homebridge Version**: 1.3.9
* **Node.js Version**: 16.13.1
* **NPM Version**: 8.3
* **Operating System**: Ubuntu 20.04 LTS

**Comment by nVuln** [comment]
Please try restart homebridge 1 or 2 times

**Comment by xcsdm** [comment]
I have tried restarting several times.  I get the same error each time.

**Comment by mpreissner** [comment]
Same issue here, plupgin version 1.2.13

**Comment by nVuln** [comment]
> I have tried restarting several times. I get the same error each time.

please enable homebridge debug mode, let see full detail of error wrote in debug logging

**Comment by xcsdm** [comment]
Here are the portions of the log referencing LGThinQ

```
[1/18/2022, 8:27:36 AM] Registering platform 'homebridge-lg-thinq.LGThinQ'
[1/18/2022, 8:27:36 AM] [LGThinQ] Initializing LGThinQ platform...
[1/18/2022, 8:27:37 AM] [LGThinQ] Executed didFinishLaunching callback
[1/18/2022, 8:27:39 AM] [LGThinQ] [request] get https://route.lgthinq.com:46030/v1/service/application/gateway-uri
[1/18/2022,` 8:27:42 AM] [LGThinQ] [request] post https://us.lgeapi.com/oauth2/token
[1/18/2022, 8:27:42 AM] [LGThinQ] LG ThinQ internal server error, try again later.
[1/18/2022, 8:27:42 AM] [LGThinQ]
[1/18/2022, 8:27:42 AM] [LGThinQ] NotConnectedError:
    at /usr/lib/node_modules/homebridge-lg-thinq/src/lib/request.ts:38:11
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Auth.getJSessionId (/usr/lib/node_modules/homebridge-lg-thinq/src/lib/Auth.ts:139:30)
    at API.refreshNewToken (/usr/lib/node_modules/homebridge-lg-thinq/src/lib/API.ts:256:5)
    at API.ready (/usr/lib/node_modules/homebridge-lg-thinq/src/lib/API.ts:239:7)
    at ThinQ.isReady (/usr/lib/node_modules/homebridge-lg-thinq/src/lib/ThinQ.ts:295:5)
```

**Comment by grovolis** [comment]
I get the same error even on older versions

**Comment by nVuln** [comment]
> I get the same error even on older versions

which country of your account?

**Comment by grovolis** [comment]
> > I get the same error even on older versions
> 
> 
> 
> which country of your account?

Account is created in the UK, however I'm currently in another country, could that be the problem?

**Comment by nVuln** [comment]
@xcsdm @grovolis please try switch to account instead of token and see if it's worked, follow config below, empty `refresh_token` and fill `username` and `password` of LG account

```json
 {
            "country": "US",
            "language": "en-US",
            "auth_mode": "account",
            "refresh_token": "<should empty it>",
            "username": "<put your username of lg account>",
            "password": "<put your password>",
            "platform": "LGThinQ",
            "devices": [],
            "thinq1": false,
            "refresh_interval": 60
        }
```

**Comment by grovolis** [comment]
> @xcsdm @grovolis please try switch to account instead of token and see if it's worked, empty `refresh_token` and fill `username` and `password` of LG account
> 
> ```json
>  {
>             "country": "US",
>             "language": "en-US",
>             "auth_mode": "account",
>             "refresh_token": "<should empty it>",
>             "username": "<put your username of lg account>",
>             "password": "<put your password>",
>             "platform": "LGThinQ",
>             "devices": [],
>             "thinq1": false,
>             "refresh_interval": 60
>         }
> ```

This worked like a charm!!! Thanks very much!!

**Comment by nVuln** [comment]
@grovolis I published `v1.2.14-test.6` to fix previous issue, can you help revert to old method and try it ?

**Comment by grovolis** [comment]
> @grovolis I published `v1.2.14-test.6` to fix previous issue, can you help revert to old method and try it ?

I will try it later today and report back

**Comment by xcsdm** [comment]
>  @grovolis I published v1.2.14-test.6 to fix previous issue, can you help revert to old method and try it ?

This test version worked for me, using the refresh token method.  Sees my washer and dryer.

Thank you
